pathway-analysis/qusage_replicate_vignette.Rmd

```diff
@@ -0,0 +1,404 @@
+```
+
+We can look at the results with the `qsTable` function.
+This will give us the top 20 pathways by default.
+
+```{r}
+qsTable(timepoint_qusage_results)
+```
+
+Let's save the results to file.
+First, let's save a nicely formatted table.
+
+```{r}
+timepoint_table <- qsTable(timepoint_qusage_results,
+                           # numPathways let's us snag all tested gene sets
+                           number = numPathways(timepoint_qusage_results))
+readr::write_tsv(timepoint_table,
+                 file.path("results", "qusage_t1-t0_kegg_results.tsv"))
+```
+
+Note that gene sets with negative fold changes are higher in `t0`.
+
+We're also going to save the `QSarray` output of `qusage` because, as we'll see
+shortly, there's lots you can do with qusage results!
+
+```{r}
+readr::write_rds(timepoint_qusage_results,
+                 file.path("results", "qusage_t1-t0_kegg_QSarray.RDS"))
+```
+
+#### Plotting
+
+We can look at the overall results with the `plotCIs` function.
+
+```{r}
+plotCIs(timepoint_qusage_results)
+```
+
+It's a bit difficult to see in this notebook, but we'll save the plot as a PDF.
+
+```{r}
+pdf(file.path("plots", "qusage_t1-t0_plotCIs.pdf"), width = 14, height = 8)
+# adjusting the figure margins to better fit pathway names
+par(mar = c(13, 4, 1, 2))
+plotCIs(timepoint_qusage_results)
+dev.off()
+```
+
+Let's look more closely at the T Cell Receptor Signaling Pathway. 
+We can see above that this pathway has a negative fold change and therefore
+should be higher at `t0`.
+
+```{r}
+pathway_index <- 
+  which(names(kegg_genesets) == "KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY")
+```
+
+Plot the overall distribution of the pathway (thick black line) and the 
+distribution of the individual genes (thin lines colored by their standard
+deviations).
+
+```{r}
+plotGeneSetDistributions(timepoint_qusage_results, path.index = pathway_index)
+```
+
+We can see the majority of genes in this pathway don't change much, but a 
+handful have lower values in `t1`.
+
+Now, plot the mean and confidence intervals for each gene in the pathway.
+This gives us insight into _which_ genes changed the most.
+
+```{r}
+plotCIsGenes(timepoint_qusage_results, path.index = pathway_index)
+```
+
+We can easily check to see if the directionality is as we expect by making
+a boxplot of the gene `387` which has a negative value and is on right side of 
+the graph.
+
+```{r}
+graphics::boxplot(exprs_mat["387", ] ~ filtered_metadata_df$timepoint)
+```
+
+### Two way comparison
+
+Patients in this influenza data set were either _symptomatic_ or _asymptomatic_.
+We can check if there are differences in pathway _responses_ to influenza virus
+(e.g., between time points) between the two groups.
+The ability to do this more complex comparison is a great feature of `qusage`.
+Again, we'll mirror the analysis performed in the vignette.
+
+The symptomatic or asymptomatic information is in the `clinic_pheno` column
```

Comment:
```suggestion
The symptomatic (`sx`) or asymptomatic (`asx`) information is in the `clinic_pheno` column
```

Reply:
The first time we use the `asx` or `sx` in variable names it's under text that says "Asymptomatic" or "Symptomatic" respectively I believe